src/azure-cli/azure/cli/command_modules/storage/_params.py

```diff
@@ -1561,6 +1561,35 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                    help='The new directory name the users want to move to. The value must have the following format: '
                         '"{filesystem}/{directory}/{subdirectory}".')
 
+    with self.argument_context('storage fs directory upload') as c:
+        from ._validators import validate_fs_directory_upload_destination_url
+        c.extra('destination_fs', options_list=['--file-system', '-f'], required=True,
+                help='The upload destination file system.')
+        c.extra('destination_path', options_list=['--destination-path', '-d'], required=True,
```

Comment:
As you defined, this parameter is required. But when they want to upload files to root directory, how should we input the value?
Could we make this one optional? If not provided, we will create directory in root like `az storage blob upload-batch`?

Reply:
also need to make sure it is not blocked by your validator

Reply:
For uploading to root directory, `-d '/'` can be passed in. I don't know whether it's ok to not align with `az storage blob upload-batch`. What do you think?

Reply:
Prefer add such support. Because in upload-batch, you could just upload to a container, not a specific path. So I hope we have such capability, too.

Reply:
done